### PR TITLE
usm: change http2_in_flight from LRU map to HASH map

### PR DESCRIPTION
**Pull request by amitslavin**
### What does this PR do?

Changs the `http2_in_flight` from LRU map to HASH map.

### Motivation

This PR replaces the LRU map to HASH map for the following reasons:
- we have identify issues in HTTP decoding by with LRU map so before releasing HTTP/2 support we would like to align with HTTP
- We have notice that the LRU map is casing some unexpected behavior while testing it in our gRPC tests

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/network/protocols/http2/protocol.go [hunk 1]
```diff
@@ -206,6 +216,36 @@ func (p *protocol) processHTTP2(data []byte) {
 	p.statkeeper.Process(tx)
 }
 
+func (p *protocol) setupMapCleaner(mgr *manager.Manager) {
+	http2Map, _, err := mgr.GetMap(inFlightMap)
+	if err != nil {
+		log.Errorf("error getting http_in_flight map: %s", err)
```
Comment (guyarb) [inline pkg/network/protocols/http2/protocol.go:222]:
```suggestion
		log.Errorf("error getting %q map: %s", inFlightMap, err)
```

File: pkg/network/protocols/http2/protocol.go [hunk 2]
```diff
@@ -206,6 +216,36 @@ func (p *protocol) processHTTP2(data []byte) {
...
+		return
+	}
+	mapCleaner, err := ddebpf.NewMapCleaner(http2Map, new(netebpf.ConnTuple), new(EbpfTx))
```
Comment (guyarb) [inline pkg/network/protocols/http2/protocol.go:225]:
```suggestion
	mapCleaner, err := ddebpf.NewMapCleaner(http2Map, new(http2StreamKey), new(EbpfTx))
```

File: pkg/network/protocols/http2/protocol.go [hunk 3]
```diff
@@ -206,6 +216,36 @@ func (p *protocol) processHTTP2(data []byte) {
...
+	if err != nil {
+		log.Errorf("error creating map cleaner: %s", err)
+		return
+	}
+
+	ttl := p.cfg.HTTP2IdleConnectionTTL.Nanoseconds()
```
Comment (guyarb) [inline pkg/network/protocols/http2/protocol.go:231]:
```suggestion
	ttl := p.cfg.HTTPIdleConnectionTTL.Nanoseconds()
```

File: pkg/network/protocols/http2/protocol.go [hunk 4]
```diff
@@ -206,6 +216,36 @@ func (p *protocol) processHTTP2(data []byte) {
...
+	mapCleaner.Clean(p.cfg.HTTP2MapCleanerInterval, func(now int64, key, val interface{}) bool {
```
Comment (guyarb) [inline pkg/network/protocols/http2/protocol.go:232]:
```suggestion
	mapCleaner.Clean(p.cfg.HTTPMapCleanerInterval, func(now int64, key, val interface{}) bool {
```

File: pkg/network/config/config.go
Comment (guyarb) [inline pkg/network/config/config.go:1]: revert

**Review by guyarb — CHANGES_REQUESTED** [review]

**Comment by val06** [comment]
can you fill the "Motivation" section of the PR?

**Review by guyarb — APPROVED** [review]

**Pull request merged**